src/components/QueryEditor/QueryBuilder/Editors/ExactValueEditor.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { QueryBuilderOperationParamEditorProps, toOption } from '@grafana/plugin-ui';
+import { InlineField, Select } from '@grafana/ui';
+
+import { getFieldValueOptions } from './utils/editorHelper';
+
+export default function ExactValueEditor(props: QueryBuilderOperationParamEditorProps) {
+  const { onChange, index, value, operation } = props;
+  const [isLoading, setIsLoading] = useState(false);
+  const [options, setOptions] = useState<SelectableValue<string>[]>([]);
+
+  return (
+    <InlineField>
+      <Select<string>
+        allowCustomValue={true}
+        allowCreateWhileLoading={true}
+        isLoading={isLoading}
+        onOpenMenu={async () => {
+          setIsLoading(true);
+          setOptions(await getFieldValueOptions(props, operation.params[0] as string));
+          setIsLoading(false);
```

Comment:
i think we can define function and use it. Also please take a look on `onChange`